filter/filter_test.go

```diff
@@ -92,6 +93,21 @@ func runCases(t *testing.T, record *zng.Record, cases []testcase) {
 		t.Run(tt.filter, func(t *testing.T) {
 			err := runTest(tt.filter, record, tt.expectedResult)
 			require.NoError(t, err)
+
+			proc, err := zql.ParseProc(tt.filter)
+			require.NoError(t, err, "filter: %q", tt.filter)
+
+			filterExpr := proc.(*ast.FilterProc).Filter
+
+			bf, err := filter.NewBufferFilter(filterExpr)
+			if assert.NoError(t, err, "filter: %q", tt.filter) &&
```

Comment:
Is it ok for filter.NewBufferFilter to ever return an error here? I can't tell from the if statement.

Reply:
It isn't. The construct is equivalent to this:
```go
assert.NoError(t, err, "filter: %q", tt.filter)
if err == nil && ... {
```
I'm happy to change it to that if you'd like.

Reply:
Would you please? It's just a nit, but I think it's easier to read.

Reply:
Done in b28d0b9db7672f402703fcf49259aaaa9dd82c27.